.github/workflows/build.yml

```diff
@@ -0,0 +1,321 @@
+name: build
+
+on:
+  push:
+    branches:
+    - master
+    tags:
+    - v*
+  pull_request:
+    branches:
+    - master
+  schedule:
+    - cron:  '23 1 * * 0'
+  release:
+    types: [published]
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group: build-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+
+  cmake-centos7:
+
+    runs-on: ubuntu-latest
+    container: gitlab-registry.cern.ch/linuxsupport/cc7-base:latest
+
+    steps:
+    - name: Install external dependencies with yum
+      run: |
+        yum update -y
+        yum install --nogpg -y \
+            cmake3 \
+            make \
+            krb5-devel \
+            libuuid-devel \
+            libxml2-devel \
+            openssl-devel \
+            systemd-devel \
+            zlib-devel \
+            devtoolset-7-gcc-c++ \
+            which \
+            python3-devel \
+            python3-setuptools \
+            git \
+            cppunit-devel
+        yum clean all
+
+    # Need to use v1 of action as image Git is too old
+    - name: Clone repository now that Git is available
+      uses: actions/checkout@v1
```

Comment:
The faster/nicer [`v2` of this action](https://github.com/actions/checkout/tree/v2.4.0) requires Git 2.18, and CentOS 7 ships `v1.8.3.1`, so fall back to the old version here which still works.